[nmstate-0.3] varlink: Fix the logger

cathay4t opened the pull request:
As https://github.com/nmstate/kubernetes-nmstate/pull/663#issuecomment-746269694
mentioned, current self-cooked json string might cause problem for
unexpected string.

Introduced `NmstateVarlinkLogHandler` to stored the `LogRecord` in a
list, and only use `NmstateVarlinkLogHandler.logs` for varlink to
generate its JSON reply.

ffmancera commented on nmstatectl/nmstate_varlink.py:
```diff
@@ -116,14 +132,14 @@ def start_varlink_server(address):
 class NmstateError(varlink.VarlinkError):
-    def __init__(self, message, string_io):
+    def __init__(self, message, logs):
         varlink.VarlinkError.__init__(
             self,
             {
                 "error": self.__class__.__name__,
                 "parameters": {
                     "error_message": message,
-                    "log": get_logs(string_io),
+                    "log": logs,
```
Black is complaining on these lines. Please check the CI. Thanks!

cathay4t committed:
varlink: Fix the logger

As https://github.com/nmstate/kubernetes-nmstate/pull/663#issuecomment-746269694
mentioned, current self-cooked json string might cause problem for
unexpected string.

Introduced `NmstateVarlinkLogHandler` to stored the `LogRecord` in a
list, and only use `NmstateVarlinkLogHandler.logs` for varlink to
generate its JSON reply.

Signed-off-by: Gris Ge <fge@redhat.com>
